caikit/runtime/http_server/http_server.py

```diff
@@ -282,11 +277,14 @@ def stop(self):
             grace_period_seconds (Union[float, int]): Grace period for service shutdown.
                 Defaults to application config
         """
-        self.server.should_exit = True
```

Comment:
Why was this moved? I think it makes sense to put it under the `if`, I'm just curious if having it outside was causing some specific problem

Reply:
Nah I was just being a bit more specific- it's only needed to signal the server running in another thread to stop